### DISCUSSION TITLE
Just hangs

### DISCUSSION THREAD

**Discussion by plettb** [VladislavAntonyuk/MauiSamples#14] [Q&A] [answered]
I downloaded and tried to open the solution on a completely clean install (both OS and VS), and the solution won't open.  VS says "Loading project 1 of 11: MauiBlazorPhotoGallery" and never gets any further than that.  I let it run like that for 24 hours and then killed VS.  Any ideas?

I tried loading the projects (*.csproj) from a clean/empty solution and that appeared to work, but there doesn't appear to be anything runnable.  Is this repository just class libraries that are used by a different launcher of some kind?

Thanks!

**Comment by VladislavAntonyuk** [comment, marked as answer]
It’s a final runnable applications. VS sometimes hangs when it tries to restore nuget packages. Clean .vs, bin, obj folders. You can try run the command: dotnet restore from the cmd.

**Reply by plettb** [reply]
I've tried that on two different machines with no success.

**Reply by VladislavAntonyuk** [reply]
what is your VS version?

**Reply by plettb** [reply]
17.2.0 Preview 5.0 (6.0.300-preview.22204.3)

**Reply by VladislavAntonyuk** [reply]
try to build a solution with the CLI and then open it with Visual Studio.

**Reply by plettb** [reply]
What command syntax would you suggest?  Which switches are important?  If I just try "devenv MauiSamples.sln /build Debug", that, too, just hangs.

**Reply by VladislavAntonyuk** [reply]
For Android
dotnet build KanbanBoard.csproj -t:Run -f net6.0-android
For iOS
dotnet build KanbanBoard.csproj -t:Run -f net6.0-ios
For macOS
dotnet build KanbanBoard.csproj -t:Run -f net6.0-maccatalyst

**Reply by plettb** [reply]
I'm sorry - you said to try building with the CLI, but I had just tried the DevEnv.  Simply running "dotnet build MauiSamples.sln" worked - the build succeeded.  However, even after a successful build, VS still just hangs.

**Reply by plettb** [reply]
Please note:  this is not a huge issue for me.  I do like to look at other people's code, but I'm sure you have better things to do.  If it works fine for you, don't worry about me.  However, I'm guessing if you tried a clean machine with a clean install of VS you'd run into the same problem.  Still, please don't feel obligated to be my tech support!

**Reply by VladislavAntonyuk** [reply]
I do not have a clean env to try, I can only suggest trying another IDE (like Rider). Do you experience the same issue if you open another project like this: https://github.com/VladislavAntonyuk/KanbanBoard

**Reply by plettb** [reply]
Just for kicks I tried it on another clean VM with a new installation of VS, and I was able to run it.  Apparently there was an unhandled exception when I tried to run the "MauiBlazorPhotoGallery" app, but that's a different problem I haven't looked into yet.

**Reply by VladislavAntonyuk** [reply]
fixed, I forgot to register MediaService.

**Reply by plettb** [reply]
Yes, it works now.  Thanks!
